performance/main.py

```diff
@@ -250,6 +250,74 @@ class ArrayGOPerfREF(ArrayGOPerf):
     entry = staticmethod(ArrayGOREF)
 
 
+#-------------------------------------------------------------------------------
+class IsNaElementPerf(Perf):
+    NUMBER = 1000
+
+    def pre(self):
+        class FloatSubclass(float): pass
+
+        self.values = [
+                # Na-elements
+                np.datetime64('NaT'), np.timedelta64('NaT'), None, float('NaN'), -float('NaN'),
+
+                # Non-float, Non-na elements
+                1, 'str', np.datetime64('2020-12-31'), datetime.date(2020, 12, 31), False,
+        ]
+
+        nan = np.nan
+        nanjs = [
+                complex(nan, 0),
+                -complex(nan, 0),
+                complex(-nan, 0),
+                -complex(-nan, 0),
+                complex(0, nan),
+                -complex(0, nan),
+                complex(0, -nan),
+                -complex(0, -nan),
```

Comment:
These aren't really giving you any more variety than you already have (although you may consider replacing them with cases where both the real and imaginary parts are signed NaNs):

```suggestion
                complex(nan, 0),
                complex(-nan, 0),
                complex(0, nan),
                complex(0, -nan),
```

Reply:
Thanks - I was trying to overly thorough here just to be extra confident I caught all different possible nanjs.